clinica/pipelines/t1_linear/t1_linear_pipeline.py

```diff
@@ -0,0 +1,258 @@
+
+    Raises:
+
+    Example:
+        >>> from t1_linear import T1Linear
+        >>> pipeline = T1Linear('~/MYDATASET_BIDS', '~/MYDATASET_CAPS')
+        >>> pipeline.parameters = {
+        >>>     # ...
+        >>> }
+        >>> pipeline.base_dir = '/tmp/'
+        >>> pipeline.run()
+    """
+
+    def check_custom_dependencies(self):
+        """Check dependencies that can not be listed in the `info.json` file.
+        """
+        pass
+
+    def get_input_fields(self):
+        """Specify the list of possible inputs of this pipeline.
+
+        Returns:
+            A list of (string) input fields name.
+        """
+
+        return ['t1w']
+
+    def get_output_fields(self):
+        """Specify the list of possible outputs of this pipeline.
+
+        Returns:
+            A list of (string) output fields name.
+        """
+
+        return ['image_id']
+
+    def build_input_node(self):
+        """Build and connect an input node to the pipeline.
+        """
+        from os import pardir
+        from os.path import dirname, join, abspath, split, exists
+        import sys
+        import nipype.interfaces.utility as nutil
+        import nipype.pipeline.engine as npe
+        from clinica.utils.inputs import check_bids_folder
+        from clinica.utils.participant import get_subject_session_list
+        from clinica.utils.exceptions import ClinicaBIDSError, ClinicaException
+        from clinica.utils.inputs import clinica_file_reader
+        from clinica.utils.input_files import T1W_NII
+        from clinica.utils.inputs import fetch_file
+        from clinica.utils.stream import cprint
+
+        check_bids_folder(self.bids_directory)
+        is_bids_dir = True
+
+        root = dirname(abspath(join(abspath(__file__), pardir, pardir)))
+        path_to_mask = join(root, 'resources', 'masks')
+        self.ref_template = join(
+                path_to_mask, 'mni_icbm152_t1_tal_nlin_sym_09c.nii')
+        self.ref_crop = join(path_to_mask, 'ref_cropped_template.nii.gz')
+        url1 = "https://aramislab.paris.inria.fr/files/data/img_t1_linear/ref_cropped_template.nii.gz"
+        url2 = "https://aramislab.paris.inria.fr/files/data/img_t1_linear/mni_icbm152_t1_tal_nlin_sym_09c.nii"
+
+        if not(exists(self.ref_template)):
+            try:
+                fetch_file(url2, self.ref_template)
+            except IOError as err:
+                cprint('Unable to download required template (mni_icbm152) for processing:', err)
+
+        if not(exists(self.ref_crop)):
+            try:
+                fetch_file(url1, self.ref_crop)
+            except IOError as err:
+                cprint('Unable to download required template (ref_crop) for processing:', err)
+
+        self.sessions, self.subjects = get_subject_session_list(
+                self.bids_directory,
+                self.tsv_file,
+                is_bids_dir,
+                False,
+                self.base_dir
+                )
```

Comment:
This is automatically done in Abstract class (engine.py)

Reply:
ok, deleted!